__tests__/api.test.js

```diff
@@ -292,14 +314,44 @@ test('ignore extra properties with a status 201', () => {
     .send(newComment)
     .expect(201)
     .then(res => {
-        expect(res.body.comment).toMatchObject({
-            comment_id: expect.any(Number),
-            body: 'This is a new comment',  
-            article_id: 1,
-            author: 'butter_bridge',
-            votes: expect.any(Number),
-            created_at: expect.any(String)
+            expect(res.body.comment).toMatchObject({
+                comment_id: expect.any(Number),
+                body: 'This is a new comment',  
+                article_id: 1,
+                author: 'butter_bridge',
+                votes: expect.any(Number),
+                created_at: expect.any(String)
+            })
         })
     })
 })
-})
+
+describe('error handling', () => {
+    test('responds with a 404 status code when given an invalid path', () => {
+        return request(app)
+        .get('/api/invalid-path')
+        .expect(404)
+        .then(res => {
+            const { msg } = res.body;
+            expect(msg).toBe('Not found');
+        })
+    })
+    test('responds with a 400 status code when given an invalid perameter', () => {
+        return request(app)
+        .get('/api/articles/string')
+        .expect(400)
+        .then(res => {
+            const { msg } = res.body;
+            expect(msg).toBe('Bad request');
+        })
+    })
+    test('responds with a 404 along without a no article found message when given an invalid article id', () => {
+        return request(app)
+        .get('/api/articles/999999')
+        .expect(404)
+        .then(res => {
+            const { msg } = res.body;
+            expect(msg).toBe('Article not found');
+        })
+    })
```

Comment:
Just one sad path test missing: inc_votes could be the wrong data type {inc_votes: 'not-a-number'}